doc/mgmt/SONiC_overlay_ConfigDB.md

```diff
@@ -0,0 +1,234 @@
+# Multiple overlay ConfigDB
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [About this Manual](#about-this-manual)
+  + [SONiC issue solved by this feature](#sonic-issue-solved-by-this-feature)
+- [1 Functional Requirement](#1-functional-requirement)
+  + [1.1 swss-common return default value from Yang model overlay](#1-1-swss-common-return-default-value-from-yang-model-overlay)
+  + [1.2 swss-common return static config from static config overlay](#1-1-swss-common-return-static-config-from-static-config-overlay)
+- [2 Design](#2-design)
+  + [2.1 Considerations](#2-1-considerations)
+  + [2.2 New class](#2-2-new-class)
+  + [2.3 Other code change](#2-3-other-code-change)
+  + [2.4 Database Schema](#2-4-database-schema)
+  + [2.5 Code example](#2-5-code-example)
+  + [2.6 Other solutions](#2-6-other-solutions)
+- [3 Error handling](#3-error-handling)
+- [4 Serviceability and Debug](#4-serviceability-and-debug)
+- [5 Unit Test](#5-unit-test)
+- [6 Migration steps](#6-migration-steps)
+  + [6.1 Phase 1](#6-1-phase-1)
+  + [6.2 Phase 1](#6-2-Phase-2)
+- [7 References](#7-references)
+
+# About this Manual
+
+This document provides a detailed description on the new features for:
+
+- Overlay Config DB.
+- Yang model default value overlay.
+- Static config overlay.
+- swss-common API change.
+
+## SONiC issue solved by this feature
+
+- Still using old default value and config from j2 template after OS upgrade:
+  - Following config may update after OS upgrade:
+    - Default value: defined in yang model.
+    - Static config: defined in j2 template.
+  - Currently all config stored in config DB, so config above can't update after OS upgrade.
+- Potential risk, Yang model default value conflict with hardcoded value:
+  - Default value hardcoded in source code.
+  - Yang model default value not used.
+- SONiC utilities not support get user config and full config:
+  - Vender OS have different show command:
+    - show running: only return user config.
+    - show running all: user config, default value from yang model, and config from j2 template.
+  - Currently SONiC only support 'show running'
+
+# 1 Functional Requirement
+
+## 1.1 swss-common return default value from Yang model overlay
+
+- Return default value is optional.
+  - Application can read config without default value, also can read config with default value.
+- Backward compatibility with existed code and applications.
+
+## 1.2 swss-common return static config from static config overlay
+
+- Static buffer config stored in static config tables.
+- Return static config is optional.
+  - Application can decide read config with static config or not.
```

Comment:
>config with static config or not [](http://example.com/codeflow?start=32&length=32)

Could you explain 2 options in detailed. Do you mean static config possible overridden by user config if existing?

Reply:
Fixed, add example.